.github/workflows/release_impl.yml

```diff
@@ -35,29 +47,53 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Prepare runner
+        uses: ./.github/actions/prepare_runner
+        with:
+          disable_ccache: true
 
       - uses: actions/download-artifact@v4
         with:
           path: release_artifacts
           pattern: clio_server_*
 
-      - name: Prepare files
+      - name: Create release notes
         shell: bash
-        working-directory: release_artifacts
         run: |
-          cp ${{ github.workspace }}/.github/workflows/${{ inputs.notes_header_file }} "${RUNNER_TEMP}/release_notes.md"
-          echo '' >> "${RUNNER_TEMP}/release_notes.md"
-          echo '```' >> "${RUNNER_TEMP}/release_notes.md"
+          echo ${{ inputs.header }} > "${RUNNER_TEMP}/release_notes.md"
+
+      - name: Generate changelog
+        shell: bash
+        if: ${{ inputs.generate_changelog }}
+        run: |
+          git-cliff
```

Comment:
Because of our bugfix releases usually I do: `git cliff --tag <current_release_tag> <last_commit_from_previous_release>..`
Otherwise `git-cliff` includes too many changes.